chain-ser/src/deser.rs

```diff
@@ -26,8 +26,18 @@ pub enum WriteError {
     IoError(#[from] std::io::Error),
 }
 
+pub trait SerializedSize {
+    fn serialized_size(&self) -> usize;
+}
+
+impl<T: SerializedSize> SerializedSize for &T {
+    fn serialized_size(&self) -> usize {
+        (*self).serialized_size()
+    }
+}
+
 /// Define that an object can be written to an `std::io::Write` object.
-pub trait Serialize {
+pub trait Serialize: SerializedSize {
```

Comment:
I'm not sure about this bound, is it really necessary? We could just request for `Serialize + SerializedSize` at some point if we need both methods.

Reply:
Actually not it is not necessary so can be update and therefore `impl SerializedSize for Ledger {` can be removed.